test/test_commands.py

```diff
@@ -207,6 +207,77 @@ def test_chrom_means(segments):
                               is_sample_female=sample_is_f)
             test_chrom_means(cns_p99)
 
+    def _get_true_copy_numbers(self):
+        """Returns a dict with the values to check against."""
+        ret = {}
+        ret['ploidy'] = 2  # TODO: test with other ploidies, too
+        ret['purity'] = 0.8
+        # Initialize 2x2 matrix for each combination of is_refence_male x is_sample_female.
+        matrix = {
+            is_reference_male: {
+                is_sample_female: [] for is_sample_female in [True, False]
+            } for is_reference_male in [True, False]
+        }
+        # Each tuple represents (row_index, chromosome_name, absolute_copies, reference_copies, expected_copies):
+        # male reference, female sample
+        matrix[True][True].append((0, "chr1", 2.24, 2, 2))
+        matrix[True][True].append((9, "chrX", 3.09, 1, 2))
+        matrix[True][True].append((10, "chrX", 5.76, 1, 2))
+        matrix[True][True].append((11, "chrX", 1.92, 1, 2))
+        matrix[True][True].append((16, "chrY", 3.38, 1, 0))
+
+        # male reference, male sample
+        matrix[True][False].append((0, "chr1", 2.24, 2, 2))
+        matrix[True][False].append((9, "chrX", 3.34, 1, 1))
+        matrix[True][False].append((10, "chrX", 6.01, 1, 1))
+        matrix[True][False].append((11, "chrX", 2.17, 1, 1))
+        matrix[True][False].append((16, "chrY", 3.13, 1, 1))
+
+        # female reference, female sample
+        matrix[False][True].append((0, "chr1", 2.24, 2, 2))
+        matrix[False][True].append((9, "chrX", 6.68, 2, 2))
+        matrix[False][True].append((10, "chrX", 12.02, 2, 2))
+        matrix[False][True].append((11, "chrX", 4.33, 2, 2))
+        matrix[False][True].append((16, "chrY", 3.38, 1, 0))
+
+        # female reference, male sample
+        matrix[False][False].append((0, "chr1", 2.24, 2, 2))
+        matrix[False][False].append((9, "chrX", 6.93, 2, 1))
+        matrix[False][False].append((10, "chrX", 12.27, 2, 1))
+        matrix[False][False].append((11, "chrX", 4.58, 2, 1))
+        matrix[False][False].append((16, "chrY", 3.13, 1, 1))  # ref_copies = 1 even for female reference
+
+        ret['matrix'] = matrix
+        return ret
+
+    def test_call_absolute_clonal_and_dataframe(self):
+        """Test some helper methods for `call`."""
```

Comment:
I added this test here before fix/refactoring `call.absolute_clonal()`/`absolute_dataframe()` so that I'm sure I'm improving things here.

Reply:
Excellent, thanks